File: scala/gradle-build-logic/java-plugins/src/main/scala/software/onepiece/gradle/pluginshowto/MyJavaLibraryPlugin.scala
package software.onepiece.gradle.pluginshowto

import org.gradle.api.{Plugin, Project}
import org.gradle.api.plugins.JavaLibraryPlugin

abstract class MyJavaLibraryPlugin extends Plugin[Project] {

    override def apply(project: Project): Unit = {
        project.getPlugins.apply(classOf[MyJavaBasePlugin])
        project.getPlugins.apply(classOf[JavaLibraryPlugin])
    }
}

File: kotlin/gradle-build-logic/java-plugins/src/main/kotlin/software/onepiece/gradle/pluginshowto/MyJavaLibraryPlugin.kt
package software.onepiece.gradle.pluginshowto

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.plugins.JavaLibraryPlugin

abstract class MyJavaLibraryPlugin : Plugin<Project> {

    override fun apply(project: Project): Unit = with(project) {
    }
}
